### PR TITLE
fix: respect transform values passed into collection.add

### LINKED ISSUE #6826: Firestore SERVER_TIMESTAMP only works with "set" and not "add"
Mac OSX Mojave 10.14
Python 3.7.0
google-cloud-firestore==0.30.0
#### Steps to reproduce

  1. Run the two code example below. `set` works but `add` doesn't. The expected behavior is that they both parse the firestore.SERVER_TIMESTAMP constant the same way.

#### Code example

this fails
```python
rows = firestore.client().collection('rows')
rows.add(
                dict(
                    created_at=firestore.SERVER_TIMESTAMP,
                    updated_at=firestore.SERVER_TIMESTAMP))
```

this works
```python
rows = firestore.client().collection('rows')
rows.document('test_id').set(
                dict(
                    created_at=firestore.SERVER_TIMESTAMP,
                    updated_at=firestore.SERVER_TIMESTAMP))
```

Seems like it was forgotten to add SERVER_TIMESTAMP parsing to the `add` method and it's only on the `set` method.

#### Stack trace
```
17:30:32 tables.1    |   File "app.py", line 1337, in mutate
17:30:32 tables.1    |     updated_at=firestore.SERVER_TIMESTAMP))
17:30:32 tables.1    |   File "env/lib/python3.7/site-packages/google/cloud/firestore_v1beta1/collection.py", line 160, in add
17:30:32 tables.1    |     fields=_helpers.encode_dict(document_data))
17:30:32 tables.1    |   File "env/lib/python3.7/site-packages/google/cloud/firestore_v1beta1/_helpers.py", line 536, in encode_dict
17:30:32 tables.1    |     for key, value in six.iteritems(values_dict)
17:30:32 tables.1    |   File "env/lib/python3.7/site-packages/google/cloud/firestore_v1beta1/_helpers.py", line 536, in <dictcomp>
17:30:32 tables.1    |     for key, value in six.iteritems(values_dict)
17:30:32 tables.1    |   File "env/lib/python3.7/site-packages/google/cloud/firestore_v1beta1/_helpers.py", line 520, in encode_value
17:30:32 tables.1    |     'Invalid type', type(value))
17:30:32 tables.1    | TypeError: ('Cannot convert to a Firestore Value', <object object at 0x104fdebe0>, 'Invalid type', <class 'object'>)
```

### PR DESCRIPTION
**Pull request by mcdonc**
closes #6826:

### CODE REVIEW

**Comment by mcdonc** [comment]
system test failure appears to be unrelated to this commit (the failing test never takes the codepath that this PR touches)

**Comment by tseaver** [comment]
Systest failures due to #7105.

File: firestore/google/cloud/firestore_v1beta1/collection.py [hunk 1]
```diff
@@ -159,9 +159,8 @@ def add(self, document_data, document_id=None):
         """
         if document_id is None:
             parent_path, expected_prefix = self._parent_info()
-            document_pb = document_pb2.Document(
-                fields=_helpers.encode_dict(document_data)
-            )
+
+            document_pb = document_pb2.Document()
```
Comment (tseaver) [inline firestore/google/cloud/firestore_v1beta1/collection.py]: Clever!  I didn't think to just use an empty document for the `create` call.

File: firestore/tests/system.py [hunk 2]
```diff
@@ -409,7 +409,13 @@ def test_collection_add(client, cleanup):
...
     snapshot1 = document_ref1.get()
     assert snapshot1.to_dict() == data1
-    assert snapshot1.create_time == update_time1
+    # NB: We used to assert here that the snapshot create_time was the document
+    # update time as returned by add(), but now under the hood, to allow for
+    # transform values passed into add, the document is created, then a set()
+    # is issued against the document later; the value returned by add is the
+    # result of the latter.  We don't have access to the create time,
+    # so we don't have a value to make this assertion with.
+    # assert snapshot1.create_time == update_time1
```
Comment (tseaver) [inline firestore/tests/system.py]: Could that assertion still be useful, but as `assert snapshot.create_time < update_time`?

File: firestore/tests/system.py [hunk 3]
```diff
@@ -429,7 +435,8 @@ def test_collection_add(client, cleanup):
     cleanup(document_ref3)
     snapshot3 = document_ref3.get()
     assert snapshot3.to_dict() == data3
-    assert snapshot3.create_time == update_time3
+    # see NB above
+    #assert snapshot3.create_time == update_time3
```
Comment (tseaver) [inline firestore/tests/system.py]: Likewise here?

Comment (mcdonc) [reply firestore/tests/system.py]: I bailed on both of these, as the Timestamp object doesn't support lteq or lt and I couldn't see the continued hard utility of making the assertion.

**Pull request merged**